tools/log-dump/run.sh

```diff
@@ -0,0 +1,340 @@
+      if [[ $STORAGE_TYPE == "jbod" ]]; then
+        for j in $(seq 0 $(($JBOD_DISKS-1))); do
+          dump_partition $TOPIC $PARTITION $i $j
+        done
+      else 
+        dump_partition $TOPIC $PARTITION $i
+      fi
+    done
+    
+  else
+    error "Missing required options"
+  fi
+}
+
+cg_offsets() {
+  if [[ -n $NAMESPACE && -n $CLUSTER && -n $GROUP_ID ]]; then
+    check_kube_conn
+    get_kafka_setup
+          
+    # dump CG offsets coordinating partition across the cluster (including replicas)
+    local partition=$(coord-partition "$GROUP_ID" "$TOT_PART")
+    echo "$GROUP_ID coordinating partition: $partition"
+    for i in $(seq 0 $(($KAFKA_BROKERS-1))); do
+      if [[ $STORAGE_TYPE == "jbod" ]]; then
+        for j in $(seq 0 $(($JBOD_DISKS-1))); do
+          dump_partition $CO_TOPIC $partition $i $j
+        done
+      else
+        dump_partition $CO_TOPIC $partition $i
+      fi
+    done
+    
+  else
+    error "Missing required options"
+  fi
+}
+
+txn_state() {
+  if [[ -n $NAMESPACE && -n $CLUSTER && -n $TXN_ID ]]; then
+    check_kube_conn
+    get_kafka_setup
+        
+    # dump TX state coordinating partition across the cluster (including replicas)
+    local partition=$(coord-partition "$TXN_ID" "$TOT_PART")
+    echo "$TXN_ID coordinating partition: $partition"
+    for i in $(seq 0 $(($KAFKA_BROKERS-1))); do
+      if [[ $STORAGE_TYPE == "jbod" ]]; then
+        for j in $(seq 0 $(($JBOD_DISKS-1))); do
+          dump_partition $TS_TOPIC $partition $i $j
+        done
+      else
+        dump_partition $TS_TOPIC $partition $i
+      fi
+    done
+    
+  else
+    error "Missing required options"
+  fi
+}
+
+cluster_meta() {
+  if [[ -n $NAMESPACE && -n $CLUSTER ]]; then
+    check_kube_conn
+    get_kafka_setup
+        
+    # dump cluster metadata partition across the cluster (including replicas)
+    local partition="0"
+    for i in $(seq 0 $(($KAFKA_BROKERS-1))); do
+      if [[ $STORAGE_TYPE == "jbod" ]]; then
+        for j in $(seq 0 $(($JBOD_DISKS-1))); do
+          dump_partition $CM_TOPIC $partition $i $j
+        done
+      else
+        dump_partition $CM_TOPIC $partition $i
+      fi
+    done
+
+  else
+    error "Missing required options"
+  fi
+}
+
+readonly USAGE="
+Usage: $0 [command] [params]
+
+  partition       Dump partition
+    --namespace     Kubernetes namespace
+    --cluster       Kafka cluster name
+    --topic         Data topic name
+    --partition     Partition number (zero-based)
+    --segment       Filter by segment name
+    --out-path      Output path (default: $OUT_PATH)
+    --dry-run       Run without dumping (default: $DRY_RUN)
+    --data          Include record payload (default: $DATA)
+
+  cg_offsets      Dump consumer group offsets by group.id
+    --namespace     Kubernetes namespace
+    --cluster       Kafka cluster name
+    --group-id      Consumer group id
+    --tot-part      Consumer offsets partitions (default: $TOT_PART)
```

Comment:
From this description I don't really know what `tot-part` is supposed to _be_.
```suggestion
    --tot-part      The number of __consumer_offsets partitions (default: $TOT_PART)
```

Reply:
ok